File: backtracking/N&M2.cpp
#include<bits/stdc++.h>
using namespace std;

int n,m;
int arr[10];

void solve(int k){
    if(k==m){
        for(int i=0;i<m;i++) cout<<arr[i]+1<<' ';
        cout<<"\n";
        return;
    }

    if(k!=0){
        for(int i=arr[k-1]+1;i<n;i++) {
            arr[k] = i;
            solve(k+1);
        }
    }
    else{
        for(int i=0;i<n;i++){
        arr[k]= i;
        solve(k+1);
        }
    }
}

int main(void){
    cin>>n>>m;
    solve(0);
}

File: backtracking/N&M2.py
import sys
input = sys.stdin.readline

def solve(k,n,m,arr):
    if k==m:
        for i in range(m):
            print(arr[i], end=" ")
        print()
    
    st = arr[k-1]+1 if k!=0 else 1
    for i in range(st,n+1):
        arr[k] = i
        solve(k+1,n,m,arr)

def main():
    n,m = map(int, input().split())
    arr = [0] * n
    solve(0,n,m,arr)

if __name__ == "__main__":
    main()
